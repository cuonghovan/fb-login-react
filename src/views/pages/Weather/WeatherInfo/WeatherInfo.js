import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Title = styled.h3`
  color: purple;
  font-size: 24px;
`;

const Metric = styled.div`
  color: black;
  font-size: 18px;
`;

const MetricContainer = styled.div`
  display: inline-block;
  text-align: left;
`;

const WeatherInfo = ({ data }) => {
  if (Object.keys(data).length === 0 && data.constructor === Object) {
    return null;
  } else {
    return (
      <div>
        <Title>Detailed weather for {data.name}</Title>
        <MetricContainer>
          <Metric>Temperature: {data.main.temp - 273.15}°C</Metric>
          <Metric>Min Temperature: {data.main.temp_min - 273.15}°C</Metric>
          <Metric>Max Temperature: {data.main.temp_max - 273.15}°C</Metric>
          <Metric>Pressure: {data.main.pressure}hpa</Metric>
          <Metric>Humidity: {data.main.humidity}%</Metric>
          <Metric>Wind: {data.wind.speed}m/s</Metric>
        </MetricContainer>
      </div>
    );
  }
};

WeatherInfo.propTypes = {
  data: PropTypes.object
};

export default WeatherInfo;
