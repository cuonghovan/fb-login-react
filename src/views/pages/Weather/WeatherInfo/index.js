export { default as WeatherInfo } from "./WeatherInfo";
export { default } from "./WeatherInfoContainer";
