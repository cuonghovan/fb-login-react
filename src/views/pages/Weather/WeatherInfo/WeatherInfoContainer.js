import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { propsToJS } from "Utils/redux-util";
import { weatherSelectors } from "State/weather";
import WeatherInfo from "./WeatherInfo";

class WeatherInfoContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    data: PropTypes.object
  };

  render() {
    return <WeatherInfo data={this.props.data} />;
  }
}

const mapStateToProps = state => ({
  data: weatherSelectors.getWeatherInfoSelect(state)
});

export default connect(mapStateToProps, null)(propsToJS(WeatherInfoContainer));
