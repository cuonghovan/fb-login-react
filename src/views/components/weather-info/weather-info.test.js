import React from 'react';
import { shallow } from 'enzyme';
import WeatherInfo from './index';

describe('weather-info', () => {

  const props = {
    data: {
      wind: {
        speed: 4.6,
        deg: 140
      },
      name: 'Hanoi',
      sys: {
        type: 1,
        id: 7980,
        message: 0.0054,
        country: 'VN',
        sunrise: 1519687115,
        sunset: 1519729216
      },
      dt: 1519722000,
      base: 'stations',
      visibility: 6000,
      weather: [
        {
          id: 803,
          main: 'Clouds',
          description: 'broken clouds',
          icon: '04d'
        }
      ],
      coord: {
        lon: 105.85,
        lat: 21.03
      },
      cod: 200,
      main: {
        temp: 295.15,
        pressure: 1011,
        humidity: 78,
        temp_min: 295.15,
        temp_max: 295.15
      },
      id: 1581130,
      clouds: {
        all: 75
      }
    }
  }

  const weatherInfo = shallow(<WeatherInfo {...props} />);

  it('renders correctly', () => {
    expect(weatherInfo).toMatchSnapshot();
  });
});
