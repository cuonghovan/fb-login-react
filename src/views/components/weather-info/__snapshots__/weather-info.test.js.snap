// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather-info renders correctly 1`] = `
<div>
  <styled.h3>
    Detailed weather for 
    Hanoi
  </styled.h3>
  <div>
    Temperature: 
    22
    °C
  </div>
  <div>
    Min Temperature: 
    22
     °C
  </div>
  <div>
    Max Temperature: 
    22
     °C
  </div>
  <div>
    Pressure: 
    1011
     hpa
  </div>
  <div>
    Humidity: 
    78
     %
  </div>
  <div>
    Wind: 
    4.6
     m/s
  </div>
</div>
`;
